src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -311,13 +310,34 @@ class LspConnectionManager {
                     * buffer of executeCommands, send to Lsp at a throttled pace (hack timeouts)  
                   * longterm: middleware? changes in Lsp?
     */
-    if (toAdd.bucket || toAdd.measurement) {
-      const payload = {bucket: toAdd.bucket?.name || this._session.bucket?.name}
-      if (toAdd.measurement) {
-        payload['measurement'] = toAdd.measurement
+    const numFieldChanges =
+      (toAdd.fields?.length || 0) + (toRemove.fields?.length || 0)
+    const numTagValueChanges =
+      (toAdd.tagValues?.length || 0) + (toRemove.tagValues?.length || 0)
+    const reInitBlock =
+      toAdd.bucket ||
+      toAdd.measurement ||
+      numFieldChanges + numTagValueChanges > 1
+
+    if (reInitBlock) {
+      const payload = {
+        bucket: toAdd.bucket?.name || this._session.bucket?.name,
+      }
+      if (toAdd.measurement || this._session.measurement) {
+        payload['measurement'] = toAdd.measurement || this._session.measurement
+      }
+      if (toAdd.fields || this._session.fields) {
+        payload['fields'] = toAdd.fields || this._session.fields
+      }
+      if (toAdd.tagValues || this._session.tagValues) {
+        payload['tagValues'] = (
+          toAdd.tagValues || this._session.tagValues
+        ).map(({key, value}) => [key, value])
```

Comment:
Just wondering, why you convert each tag-value object into an array? Is that because LSP expects array instead of object?

Reply:
exactly.
https://github.com/influxdata/flux-lsp/blob/master/src/server/commands.rs#L163